frontend/internal-packages/agent/src/langchain/agents/pmAnalysisAgent/agent.ts

```diff
@@ -1,38 +1,116 @@
 import { type BaseMessage, SystemMessage } from '@langchain/core/messages'
+import type { Runnable } from '@langchain/core/runnables'
 import { ChatOpenAI } from '@langchain/openai'
-import { toJsonSchema } from '@valibot/to-json-schema'
 import * as v from 'valibot'
 import { PM_ANALYSIS_SYSTEM_MESSAGE } from './prompts'
 
-const requirementsAnalysisSchema = v.object({
+// Direct JsonSchema definition instead of using toJsonSchema
+// because the generated schema has subtle incompatibilities with withStructuredOutput
+// (specifically the properties:{}, required:[] structure).
+// TODO: Migrate from valibot to zod, which is officially supported by langchain
+const REQUIREMENTS_ANALYSIS_SCHEMA = {
+  title: 'RequirementsAnalysis',
+  type: 'object',
+  properties: {
+    businessRequirement: { type: 'string' },
+    functionalRequirements: {
+      type: 'object',
+      properties: {},
+      required: [],
+      additionalProperties: {
+        type: 'array',
+        items: { type: 'string' },
+      },
+    },
+    nonFunctionalRequirements: {
+      type: 'object',
+      properties: {},
+      required: [],
+      additionalProperties: {
+        type: 'array',
+        items: { type: 'string' },
+      },
+    },
+  },
+  required: [
+    'businessRequirement',
+    'functionalRequirements',
+    'nonFunctionalRequirements',
+  ],
+  additionalProperties: false,
+}
+
+const requirementsAnalysisSchema = v.strictObject({
   businessRequirement: v.string(),
   functionalRequirements: v.record(v.string(), v.array(v.string())),
   nonFunctionalRequirements: v.record(v.string(), v.array(v.string())),
 })
-
 type AnalysisResponse = v.InferOutput<typeof requirementsAnalysisSchema>
 
+const reasoningSchema = v.object({
+  type: v.literal('reasoning'),
+  summary: v.array(
+    v.object({
+      type: v.literal('summary_text'),
+      text: v.string(),
+    }),
+  ),
+})
+type Reasoning = v.InferOutput<typeof reasoningSchema>
+
+type RunInput = (BaseMessage | SystemMessage)[]
+
+type RunOutput = AnalysisResponse
+
+type AnalysisWithReasoning = {
+  response: AnalysisResponse
+  reasoning: Reasoning | null
+}
+
 export class PMAnalysisAgent {
-  private analysisModel: ReturnType<ChatOpenAI['withStructuredOutput']>
+  private analysisModel: Runnable<
+    RunInput,
+    {
+      raw: BaseMessage
+      parsed: RunOutput
+    }
+  >
 
   constructor() {
     const baseModel = new ChatOpenAI({
       model: 'o4-mini',
+      reasoning: { effort: 'high', summary: 'detailed' },
```

Comment:
Effort should be tuned depending on the application, but I thought that should be done only after the bench markers are ready 👍🏻